lib/Controller/ObjectsController.php

```diff
@@ -0,0 +1,359 @@
+            
+            // Get extend parameter if present
+            $extend = $data['extend'] ?? $data['_extend'] ?? [];
+            if (is_string($extend)) {
+                $extend = array_map('trim', explode(',', $extend));
+            }
+
+            // Ensure ID in data matches URL parameter
+            $data['id'] = $id;
+
+            // Save the updated object
+            $object = $this->objectService->saveObject(objectType: $objectType, object: $data, extend: $extend);
+            
+            // Return the updated object as a JSON response
+            return new JSONResponse($object);
+        } catch (Exception $e) {
+            return new JSONResponse(
+                ['error' => $e->getMessage()],
+                400
+            );
+        }
+	}
+
+	/**
+	 * Delete an object
+	 *
+	 * @NoAdminRequired
+	 * @NoCSRFRequired
+	 *
+	 * @return JSONResponse
+	 */
+	public function destroy(string $objectType, string $id): JSONResponse
+	{
+        try {
+            // Delete the object
+            $result = $this->objectService->deleteObject($objectType, $id);
+
+            // Return the result as a JSON response
+            return new JSONResponse(['success' => $result], $result === true ? 200 : 404);
+        } catch (Exception $e) {
+            return new JSONResponse(
+                ['error' => $e->getMessage()],
+                400
+            );
+        }
+	}
+
+	/**
+     * Get audit trail for a specific object
+     *
+     * @NoAdminRequired
+     * @NoCSRFRequired
+	 *
+	 * @return JSONResponse
+     */
+    public function getAuditTrail(string $objectType, string $id): JSONResponse
+    {
+        try {
+
+            $auditTrail = $this->objectService->getAuditTrail($objectType, $id);
+            return new JSONResponse($auditTrail);
+        } catch (Exception $e) {
+            return new JSONResponse(
+                ['error' => $e->getMessage()],
+                400
+            );
+        }
+    }
+
+    /**
+     * Get all relations for a specific object
+     *
+     * @NoAdminRequired
+     * @NoCSRFRequired
+     *
+     * @return JSONResponse
+     */
+    public function getRelations(string $objectType, string $id): JSONResponse
+    {
+        try {
+            // Fetch the object by its ID
+            $relations = $this->objectService->getRelations($objectType, $id);
+
+            // Return the object as a JSON response
+            return new JSONResponse($relations);
+        } catch (Exception $e) {
+            return new JSONResponse(
+                ['error' => $e->getMessage()],
+                400
+            );
+        } 
+    }
+
+    /**
+     * Get all uses for a specific object
+     *
+     * @NoAdminRequired
+     * @NoCSRFRequired
+     *
+     * @return JSONResponse
+     */
+    public function getUses(string $objectType, string $id): JSONResponse
+    {
+        $uses = $this->objectService->getUses($objectType, $id);
```

Comment:
ObjectService does not have a function getUses()

Reply:
Still needs to be fixed